docs-website/CONTRIBUTING.md

```diff
@@ -1,121 +1,178 @@
-# Contributing to Haystack Docs
+# Contributing to Haystack Documentation
 
-This guide explains how to contribute to the Haystack documentation in this repository. It complements the main Haystack contribution guide in `deepset-ai/haystack` and focuses on authoring, reviewing, and maintaining docs.
+Thank you for your interest in contributing to the Haystack documentation! This guide provides everything you need to write, review, and maintain high-quality documentation for the Haystack project.
 
-If you plan to contribute code, tests, or integration changes, see the [main contribution guide](https://github.com/deepset-ai/haystack/blob/main/CONTRIBUTING.md) in the Haystack repository instead. For docs-related PRs that touch both code and docs, please follow both guides.
+This guide focuses specifically on documentation contributions. For code contributions, tests, or integrations in the main Haystack codebase, see the [main Haystack contribution guide](https://github.com/deepset-ai/haystack/blob/main/CONTRIBUTING.md).
 
-## What you’ll need
+## TL;DR — Your first docs PR in 10 minutes
 
-- Node.js 18+ and npm (recommended)
-- Optional: [Vale](https://github.com/errata-ai/vale) for prose/style linting. See [Prose/style linting with Vale](#prose-style-linting-with-vale) section for more details.
-
-Install Node.js and npm (macOS examples):
-
-- Recommended (using nvm): see [nvm docs](https://github.com/nvm-sh/nvm#installing-and-updating)
+1. Clone and enter the docs site:
 
 ```bash
-brew install nvm
-mkdir -p ~/.nvm
-echo 'export NVM_DIR="$HOME/.nvm"' >> ~/.zshrc
-echo '[ -s "/opt/homebrew/opt/nvm/nvm.sh" ] && . "/opt/homebrew/opt/nvm/nvm.sh"' >> ~/.zshrc
-exec $SHELL -l
-nvm install 18
-nvm use 18
-node -v && npm -v
+git clone https://github.com/YOUR_USERNAME/haystack.git
+cd haystack/docs-website
 ```
 
-- Alternative (Homebrew or official installer):
+2. Install and start:
 
 ```bash
-brew install node
+npm install
+npm start
 ```
 
-Or download an installer from the Node.js website: [nodejs.org/download](https://nodejs.org/en/download)
+**Note:** All subsequent commands in this guide should be run from the `haystack/docs-website` directory unless otherwise specified.
 
-Start a local dev server:
+3. Edit under `docs/` for the unstable version, and under `versioned_docs/version-<highest>/` for the latest stable release. If you add a new page, include its ID in `sidebars.js` or the appropriate versioned sidebar.
+
+4. Optional production check:
```

Comment:
here we say this is optional but later "Before opening a PR, ensure the site builds cleanly:"
What's the right thing to do according to you? Building takes some time....